symengine/pow.cpp

```diff
@@ -164,6 +164,40 @@ RCP<const Basic> pow(const RCP<const Basic> &a, const RCP<const Basic> &b)
         RCP<const Pow> A = rcp_static_cast<const Pow>(a);
         return pow(A->get_base(), mul(A->get_exp(), b));
     }
+    if (eq(*a, *E)) {
+        if (is_a<Mul>(*b)
+            and is_a_Complex(*down_cast<const Mul &>(*b).get_coef())) {
+            const map_basic_basic &dict = down_cast<const Mul &>(*b).get_dict();
+            RCP<const Number> coef = down_cast<const Mul &>(*b).get_coef();
+            if (dict.size() == 1) {
+                if ((is_a<Integer>(*mul(coef, I))
+                     or (is_a<Rational>(*mul(coef, I))
+                         and (eq(*mul(coef, I), *rational(1, 2))
+                              or eq(*mul(coef, I), *rational(-1, 2)))))) {
+                    for (const auto &p : dict) {
+                        if (eq(*p.first, *pi) and eq(*p.second, *one)) {
+                            return add(cos(mul(mul(I, minus_one), b)),
+                                       mul(I, sin(mul(mul(I, minus_one), b))));
+                        }
+                    }
+                }
+            }
+        } else if (is_a<Add>(*b)) {
+            const umap_basic_num &dict = down_cast<const Add &>(*b).get_dict();
+            umap_basic_num new_dict;
+            RCP<const Number> coef = down_cast<const Add &>(*b).get_coef();
+            RCP<const Basic> s = one;
+            for (const auto &p : dict) {
+                if (eq(*p.first, *pi) and is_a_Complex(*p.second)) {
+                    s = exp(mul(p.first, p.second));
```

Comment:
Can you take out the logic of mul to a new function and then call it from here so that we know if `s` was simplified that it is a number, else we can do `Add::dict_add_term(new_dict, p.second, p.first);`